### PR TITLE
Add ABCI app api to the spec

### PR DESCRIPTION
**Pull request by mossid**
Including querying functions, IBC packet format, and Msg types

### CODE REVIEW

File: spec/peg_zone_spec.md [hunk 1]
```diff
@@ -77,19 +77,19 @@ it sends a signed message to the ABCI app.
 on the Cosmos side. It locks the received funds to the consensus of the peg
 zone, logging an event that notifies the relayers.
 1. The relayers process connected via RPC to an Ethereum full node, listening for `Lock` event.
-1. Once the node receives a deposit to the smart contract it waits for 100 blocks (finality threshold) and then generates and signs a transactions that attests witness to the event
+1. Once the node receives a deposit to the smart contract it waits for 100 blocks (finality threshold) and then generates and signs a `SignIncomingMsg` that attests witness to the event
```
Comment (adrianbrink) [inline spec/peg_zone_spec.md]: How about WitnessMsg as a name?

File: spec/peg_zone_spec.md [hunk 2]
```diff
@@ -77,19 +77,19 @@ it sends a signed message to the ABCI app.
...
 to which the Cosmos peg zone is listening.
 1. The peg zone receives witness transactions until a super-majority of the voting power has witnessed an event. Every BeginBlock invocation the peg zone checks whether any incoming Ethereum transfers have reached a super-majority of confirmations.
-1. The node then updates the state with an internal transaction to reflect that someone wants to send tokens from Ethereum and generates IBC packet to mint the tokens to specified destination chain.
+1. The node then updates the state with an internal transaction to reflect that someone wants to send tokens from Ethereum and generates `IBCIncoming` to mint the tokens to specified destination chain.
```
Comment (adrianbrink) [inline spec/peg_zone_spec.md]: This should be `IBCOutgoing`.

File: spec/peg_zone_spec.md [hunk 3]
```diff
@@ -77,19 +77,19 @@ it sends a signed message to the ABCI app.
...
 
 ## Sending Ethereum tokens from Cosmos to Ethereum
 
 ![Cosmos to Ethereum](./pegzone-to-ether.jpg)
 
-1. The ABCI app receives an IBC packet that requests for burning Ethereum tokens and handles it according to the IBC specification. The ABCI app generates a valid Ethereum transaction containing {address, token address, amount, nonce}, and writes it to its state.
+1. The ABCI app receives an `IBCOutgoing` that requests for burning Ethereum tokens and handles it according to the IBC specification. The ABCI app generates a valid Ethereum transaction containing {address, token address, amount, nonce}, and writes it to its state.
```
Comment (adrianbrink) [inline spec/peg_zone_spec.md]: `IBCIncoming`.

File: spec/peg_zone_spec.md [hunk 4]
```diff
@@ -77,19 +77,19 @@ it sends a signed message to the ABCI app.
...
 1. Each signing app is watching for new transactions in the ABCI state, and detects the new transaction. 
 1. Each signing app signs the transaction using secp256k1 using a key that is known to the Ethereum smart contracts.
-1. Each signing app submits their signatures back to the ABCI app for replication.
+1. Each signing app submits their signatures back to the ABCI app as `SignOutgoingMsg` for replication.
```
Comment (adrianbrink) [inline spec/peg_zone_spec.md]: How about `SignatureMsg`.

File: spec/peg_zone_spec.md [hunk 5]
```diff
@@ -99,11 +99,11 @@ to which the Cosmos peg zone is listening.
 
 ![Cosmos to Ethereum](./pegzone-to-ether.jpg)
 
-1. the ABCI app receives an IBC packet from the hub that requests for locking Cosmos tokens and handles it according to the IBC specification. The ABCI app generates a valid Ethereum transaction containing {address, denomination, amount, nonce}, and writes it to its state. 
+1. the ABCI app receives an `IBCOutgoing` from the hub that requests for locking Cosmos tokens and handles it according to the IBC specification. The ABCI app generates a valid Ethereum transaction containing {address, denomination, amount, nonce}, and writes it to its state.
```
Comment (adrianbrink) [inline spec/peg_zone_spec.md]: Actually all these are `IBCMsg`. They are just posted into `IncomingQueue` and `OutgoingQueue`.

Comment (mossid) [reply spec/peg_zone_spec.md]: The naming I used here is bit ambiguous. In this api, all 'Incoming's means that is coming from the Ethereum contract, and all 'Outgoing's means that is going to the Ethereum contract. It confuses with IBC perspective, so changing them to 'Witness' and 'Signature' will solve this situation.

**Pull request merged**